### DISCUSSION TITLE
After installing the Vercel integration for Supabase, it only lets me create a new Project. There's no way to use an existing Project. Why?

### DISCUSSION THREAD

**Discussion by j0b0sapi3n** [supabase/supabase#29445] [Questions]
I'm trying to deploy this NextJS x Supabase boilerplate: https://github.com/KolbySisk/next-supabase-stripe-starter

I've set up a Supabase project already based on the instructions. When I try to Deploy to Vercel via the button, it asks me to install the Vercel integration for my Supabase organization which seems to work.

However, it then asks me to create a new project on Supabase. I don't see anywhere to use an existing project.
<img width="775" alt="Screenshot 2024-09-22 at 10 14 11 PM" src="https://github.com/user-attachments/assets/6b8aa11d-26bf-47d0-87ea-e09a1bd0f028">

I tried a different [NextJS Supabase boilerplate](https://github.com/vercel/nextjs-subscription-payments) and it's the same thing. I always get stuck at this step.

What am I doing wrong? Do I not have the right permissions somewhere?

**Comment by WorkingJB** [comment]
@j0b0sapi3n so you actually need to do this from within the Vercel project. It's not super obvious but within your project Settings, go to Integrations and then Browse Marketplace

![Screenshot 2025-01-20 at 07 56 31](https://github.com/user-attachments/assets/d4adbecd-9bbc-4092-bce4-4e1fad002aa2)

Go to the Supabase option, then select the ... next to Install and select Connect Account:
![Screenshot 2025-01-20 at 07 56 41](https://github.com/user-attachments/assets/ade4514c-190a-4aa6-9660-5f8df5370934)

Then you can select your existing project and connect it to your Vercel instance:
![Screenshot 2025-01-20 at 07 56 54](https://github.com/user-attachments/assets/9f36190d-a1c7-41e9-a771-3c6197c54c97)


Hope that helps!

**Comment by flogy** [comment]
For me it does not even show the "Connect Account" option when clicking the "..." button but only a single "Deploy Template" option. Maybe it is because I've already connected a Supabase project but in another Vercel project an another Supabase organization?

At this point I'm not sure if it is possible to connect project A and project B (both within a single Vercel account) to project A in organization A and project B in organization B in Supabase (also both organizations were created with the same Supabase account)... Has anyone done this before and can give me a hint? Thanks :)

**Reply by mh550** [reply]
Looks like they changed the UI. I was able to press "Deploy Template" after following the steps above until there, and I was given an option to connect an existing account or create a new account. Hope you figured it out.

**Reply by flogy** [reply]
Oh, yes, now I got it working by first going to the specific Vercel project -> Settings -> Integrations -> Browse Marketplace -> choose "Supabase (Storage)" -> Install -> Link Existing Supabase Account 🥳 There, I was able to select my Vercel and Supabase project and link them together correctly. No idea why I could not find it before.

**Reply by runbco** [reply]
hi @mh550 , any idea how i do the same but with a different organisation (in supabase). I cannot find a way to select/see an organisation that was created via supabase. 
Thank you in advance.

**Reply by conca** [reply]
Unfortunately this workaround doesn't seem to work for v0

**Reply by runbco** [reply]
yeah, i spent 2 days on trying to figure it out. I am able to add my environment variables (and it works) but v0 constantly wants me to configure supabase (the warning is very annoying).

**Comment by crfpr** [comment]
So, it seems that you can use Supabase integrations added via the Marketplace as @WorkingJB illustrated, without needing to install the "required" v0 administered integration. So far this is working okay in my project.

<img width="453" height="651" alt="Screenshot 2025-11-10 at 10 07 55" src="https://github.com/user-attachments/assets/94e76b7c-1a4a-42a7-8293-be80be3b477b" />

As long as you sync the environment variables in Supabase, it should work, but I guess we lose out on features like v0 running SQL automatically in Supabase.

Hopefully Vercel/v0 can simplify this a bit and streamline the competing integrations and not-really-required-but-still-labelled-required integration.
